Modules/_datetimemodule.c

```diff
@@ -1832,11 +1832,18 @@ wrap_strftime(PyObject *object, PyObject *format, PyObject *timetuple,
               PyObject *tzinfoarg)
 {
     PyObject *result = NULL;            /* guilty until proved innocent */
+    PyObject *strftime = NULL;          /* time.strftime */
 
     PyObject *zreplacement = NULL;      /* py string, replacement for %z */
     PyObject *colonzreplacement = NULL; /* py string, replacement for %:z */
     PyObject *Zreplacement = NULL;      /* py string, replacement for %Z */
     PyObject *freplacement = NULL;      /* py string, replacement for %f */
+#ifdef NORMALIZE_CENTURY
+    PyObject *year_str;                 /* py string, year */
+    PyObject *year;                     /* py int, year */
+    long year_long;                     /* year as long int */
+    char year_formatted[12];            /* formatted year with century for %Y/G */
```

Comment:
This is not enough for 64-bit long.

I suggest to use `SIZEOF_LONG*5/2+2`.

Reply:
Such a neat way of calculating the size of buffer needed. Thanks!